sylvia-derive/src/parser/attributes/override_entry_point.rs

```diff
@@ -0,0 +1,97 @@
+use proc_macro2::TokenStream;
+use proc_macro_error::emit_error;
+use quote::quote;
+use syn::parse::{Error, Parse, ParseStream, Parser};
+use syn::spanned::Spanned;
+use syn::{parenthesized, Attribute, Ident, Path, Result, Token, Type};
+
+use crate::crate_module;
+use crate::parser::MsgType;
+
+#[derive(Debug, Clone)]
+pub struct OverrideEntryPoint {
+    entry_point: Path,
+    msg_name: Type,
+    msg_type: MsgType,
+}
+
+impl OverrideEntryPoint {
+    pub fn new(attr: &Attribute) -> Result<Self> {
+        attr.meta
+            .require_list()
+            .and_then(|meta| OverrideEntryPoint::parse.parse2(meta.tokens.clone()))
+            .map_err(|err| {
+                emit_error!(attr.span(), err);
+                err
+            })
+    }
+
+    pub fn emit_multitest_dispatch(&self) -> TokenStream {
+        let Self {
+            entry_point,
+            msg_name,
+            msg_type,
+            ..
+        } = self;
+
+        let sylvia = crate_module();
+        let values = msg_type.emit_ctx_values();
+
+        quote! {
+            #entry_point ( #values .into(), #sylvia ::cw_std::from_json::< #msg_name >(&msg)?)
+                .map_err(Into::into)
+        }
+    }
+}
+
+pub trait FilteredOverrideEntryPoints {
+    fn get_entry_point(&self, ty: MsgType) -> Option<&OverrideEntryPoint>;
+}
```

Comment:
I think that this functionality is so generic that it might be included in the type behavior, but we can think about it outside of this PR.